libtenzir/src/tql2/eval_binary.cpp

```diff
@@ -434,6 +434,41 @@ struct EvalBinOp<ast::binary_op::in, string_type, string_type> {
   }
 };
 
+// TODO: Compare numbers across types
+// TODO: {} in [{}, ...]
+template <basic_type L>
+struct EvalBinOp<ast::binary_op::in, L, list_type> {
```

Comment:
Did you consider implementing this in terms of `EvalBinOp<ast::binary_op::equal, …>`?

I think you can achieve this by slicing the left-hand side to single-element arrays, and getting the corresponding value-array from  the right-hand side for each entry (assuming it is not null or empty).

Reply:
You probably could do that, but then you need a build a long if-chain to dispatch the correct list type. Secondly, you'd need to copy (allocate) the output arrays into a single one.